src/core/exec-invoke.c

```diff
@@ -3619,18 +3619,24 @@ static int apply_working_directory(
                 const ExecContext *context,
                 const ExecParameters *params,
                 ExecRuntime *runtime,
-                const char *home) {
+                const char *pwent_home,
+                const char *const *env) {
 
         const char *wd;
         int r;
 
         assert(context);
 
         if (context->working_directory_home) {
-                if (!home)
-                        return -ENXIO;
-
-                wd = home;
+                /* Preferably use the data from $HOME, in case it was updated by a PAM module */
+                wd = strv_env_get((char**) env, "HOME");
```

Comment:
Why the cast? strv_env_get takes char* const* just fine?

Reply:
I requested for char * const * though...

Reply:
-> #36303.